provider/datagen/src/lib.rs

```diff
@@ -248,6 +249,7 @@ pub fn keys_from_file<P: AsRef<Path>>(path: P) -> std::io::Result<Vec<DataKey>>
 /// # Ok(())
 /// # }
 /// ```
+#[deprecated]
 pub fn keys_from_bin<P: AsRef<Path>>(path: P) -> std::io::Result<Vec<DataKey>> {
```

Comment:
Question: don't we still need the keyextract API? It is `#[deprecated]` with no replacement

Reply:
We're restricted by the return type on this. There should be a new `config_from_bin` or something (or we redesign config so that we return some sub-struct).